packages/commerce-server/src/prisma-next-serializer.ts

```diff
@@ -8,6 +8,8 @@ export function convertToSerializeForNextResponse(result: any) {
       result[resultKey] = result[resultKey].toISOString()
     } else if (result[resultKey]?.constructor?.name === 'Decimal') {
       result[resultKey] = result[resultKey].toNumber()
+    } else if (result[resultKey] instanceof Object) {
+      result[resultKey] = convertToSerializeForNextResponse(result[resultKey])
```

Comment:
Added this recursive case for when the Prisma object to be serialized had included a relation, the nested relation object needs to be appropriately serialized as well.